src/app/beer_garden/events/events_manager.py

```diff
@@ -1,59 +1,107 @@
 
-def set_upstream(queue: Queue) -> None:
-    global upstream_queue
-    upstream_queue = queue
 
+class EventManagerBase:
+    """Base event manager"""
 
-def publish_event(event_type):
-    # TODO - This is kind of gross
-    @wrapt.decorator(enabled=lambda: not getattr(beer_garden, "_running_tests", False))
-    def wrapper(wrapped, _, args, kwargs):
-        event = Event(name=event_type.name, payload="", error=False)
+    def do_publish(self, event):
+        pass
+
+    def create_event(self, name, payload, error, args=None, kwargs=None):
+        # TODO - We really need to standardize what an event looks like
+
+        event = Event(name=name, payload=payload, error=error)
+
+        if error:
+            # here the payload is an exception, so just stringify it
+            event.payload = str(payload)
 
-        try:
-            result = wrapped(*args, **kwargs)
-        except Exception as ex:
-            event.error = True
-            event.payload = str(ex)
-            raise
         else:
-            if event.name in (
-                Events.INSTANCE_INITIALIZED.name,
-                Events.INSTANCE_STARTED.name,
-                Events.INSTANCE_STOPPED.name,
-                Events.INSTANCE_UPDATED.name,
-                Events.REQUEST_CREATED.name,
-                Events.REQUEST_STARTED.name,
-                Events.REQUEST_COMPLETED.name,
-                Events.SYSTEM_CREATED.name,
-            ):
-                event.payload = result
-            elif event.name in (
-                Events.REQUEST_UPDATED.name,
-                Events.SYSTEM_UPDATED.name,
-            ):
-                event.payload = result
+            if event.name in (Events.REQUEST_UPDATED.name, Events.SYSTEM_UPDATED.name):
                 event.metadata = args[1]
             elif event.name in (Events.QUEUE_CLEARED.name, Events.SYSTEM_REMOVED.name):
                 event.payload = {"id": args[0]}
-            elif event.name in (Events.DB_CREATE.name, Events.DB_UPDATE.name):
-                event.payload = result
             elif event.name in (Events.DB_DELETE.name,):
                 event.payload = args[0]
-        finally:
-            upstream_queue.put(event)
 
-        return result
+        self.do_publish(event)
+
+
+class MainEventManager(EventManagerBase):
+    """Will be used to appropriately route events.
+
+    This will be the main process's manager.
+    """
+
+    def __init__(self):
+        self.forward_queues = []
+        self.callbacks = {
+            Events.INSTANCE_STOPPED.name: [
+                lambda x: print("Instance Stopped", file=sys.stderr)
+            ]
+        }
+
+    def register_forward(self, new_queue):
+        self.forward_queues.append(new_queue)
+
+    def register_callback(self, event_name: str, callback: Callable):
+        if event_name not in self.callbacks:
+            self.callbacks[event_name] = []
+        self.callbacks[event_name].append(callback)
```

Comment:
When we are registering the callback, should it be a global declaration or function based?
Example: 
All request objects get ran the same way for callbacks
```
@publish_event(Events.REQUEST_CREATE)
def create_request(request):
``` 

Example 2:
Only events kicked off by this annotation get processed by ```do_function```
```
@publish_event(Events.REQUEST_CREATE, callback=do_function)
def create_request(request):
```